org.sonarlint.eclipse.cdt/src/org/sonarlint/eclipse/cdt/internal/CProjectConfigurator.java

```diff
@@ -0,0 +1,120 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.eclipse.cdt.core.CCorePlugin;
+import org.eclipse.cdt.core.model.CoreModel;
+import org.eclipse.cdt.core.model.ICProject;
+import org.eclipse.cdt.core.parser.IScannerInfo;
+import org.eclipse.cdt.core.parser.IScannerInfoProvider;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.sonarlint.eclipse.core.configurator.ProjectConfigurationRequest;
+import org.sonarlint.eclipse.core.configurator.ProjectConfigurator;
+import org.sonarlint.eclipse.core.internal.SonarLintCorePlugin;
+import org.sonarsource.sonarlint.core.client.api.util.FileUtils;
+
+public class CProjectConfigurator extends ProjectConfigurator {
+  private static final String BUILD_WRAPPER_OUTPUT_PROP = "sonar.cfamily.build-wrapper-output";
+  private static final String BUILD_WRAPPER_OUTPUT_DIRNAME = "bw-outputs";
+  private static final String BUILD_WRAPPER_OUTPUT_FILENAME = "build-wrapper-dump.json";
+  private static final Charset BUILD_WRAPPER_OUTPUT_CHARSET = StandardCharsets.UTF_8;
+  private final BuildWrapperJsonFactory jsonFactory;
+
+  public CProjectConfigurator() {
+    this.jsonFactory = new BuildWrapperJsonFactory();
+  }
+
+  @Override
+  public boolean canConfigure(IProject project) {
+    return SonarCdtPlugin.hasCNature(project);
+  }
+
+  @Override
+  public void configure(ProjectConfigurationRequest request, IProgressMonitor monitor) {
+    ICProject cProject = CoreModel.getDefault().create(request.getProject());
+
+    Collection<IFile> filesToAnalyze = request.getFilesToAnalyze()
+      .stream().filter(CoreModel::isTranslationUnit)
+      .collect(Collectors.toList());
+
+    try {
+      Path jsonPath = configureCProject(cProject, filesToAnalyze, jsonFactory);
+      SonarLintCorePlugin.getDefault().debug("Wrote build info to: " + jsonPath.toString());
+      cleanCppCache(cProject.getProject());
+      request.getSonarProjectProperties().put(BUILD_WRAPPER_OUTPUT_PROP, jsonPath.getParent().toString());
+    } catch (Exception e) {
+      SonarLintCorePlugin.getDefault().error(e.getMessage(), e);
+    }
+  }
+
+  private static Path getProjectBaseDir(IProject project) {
+    IPath projectLocation = project.getLocation();
+    // In some infrequent cases the project may be virtual and don't have physical location
+    return projectLocation != null ? projectLocation.toFile().toPath() : ResourcesPlugin.getWorkspace().getRoot().getLocation().toFile().toPath();
+  }
+
+  private static void cleanCppCache(IProject project) {
+    IPath projectSpecificWorkDir = project.getWorkingLocation(SonarLintCorePlugin.PLUGIN_ID);
+    Path cppCacheDir = projectSpecificWorkDir.toFile().toPath().resolve("cfamily").resolve("pph");
+    if (Files.exists(cppCacheDir)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this with a call to the "toFile().exists" method [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3725)